File: tests/corpora/reflection/categories/mutation-injected/js/0014-js-function-constructor/source/calculator.js
// Clean closed-world base - synthetic, no reflection / dynamic dispatch.
// Used as an injection target by pipeline/inject_reflection.py.
// Ground truth (pre-injection): closed-world.

function add(a, b) {
  return a + b;
}

function subtract(a, b) {
  return a - b;
}

function multiply(a, b) {
  return a * b;
}

function run(x, y) {
  const s = add(x, y);
  const f = new Function("a", "b", "return a + b;");
  f(x, y);
  const d = subtract(x, y);
  return multiply(s, d);
}

module.exports = { add, subtract, multiply, run };

File: tests/corpora/reflection/clean_bases/python/calculator.py
# Clean closed-world base - synthetic, no reflection / dynamic dispatch.
# Used as an injection target by pipeline/inject_reflection.py.
# Ground truth (pre-injection): closed-world.


def add(a, b):
    return a + b


def subtract(a, b):
    return a - b


def multiply(a, b):
    return a * b


def run(x, y):
    s = add(x, y)
    d = subtract(x, y)
    return multiply(s, d)
